xbmc/guilib/GUIControlGroupList.h

```diff
@@ -64,6 +69,42 @@ class CGUIControlGroupList : public CGUIControlGroup
 
   int GetNumItems() const;
   int GetSelectedItem() const;
+  /*!
+   * \brief Returns the position of the specified child control
+   * \param control The control to be located.
+   * \return Position of the control.
+   *         -1 for a control that is not visible or cannot be found.
+   */
+  float GetControlOffset(CGUIControl* control) const;
+  /*!
+   * \brief Locate a visible and focusable child control at/close to the specified position
+   * \param target Ideal position of the control
+   * \param direction >0: Favor the control at or after the target for direction, otherwise
+   *        the control before or at target.
+   * \return Best match control & position.
+   *         The control may be null and the offset 0 if there aren't any visible focusable controls in the list.
+   */
+  SGUIControlAndOffset GetFocusableControlAt(float target, int direction) const;
+  /*!
+   * \brief Scroll the list of controls, update the selected control and the pager.
+   * \param pages direction and amount of scrolling.
+   *        For example 1 to scroll down/right one page, -0.5 to scroll up/left half a page
+   */
+  void ScrollPages(float pages);
+  /*!
+   * \brief Change the selected control and scroll to the specified position.
+   * \param control New selected control
+   * \param offset Offset of the top of the visible view
+   */
+  void MoveTo(CGUIControl* control, float offset);
+  /*!
+   * Return the first visible and focusable child control.
+   */
+  CGUIControl* GetFirstFocusableControl() const;
```

Comment:
```
xbmc/guilib/GUIControlGroupList.h:103:16: warning: 'CGUIControlGroupList::GetFirstFocusableControl' hides overloaded virtual function [-Woverloaded-virtual]
  CGUIControl* GetFirstFocusableControl() const;
               ^
xbmc/guilib/GUIControlGroup.h:58:24: note: hidden overloaded virtual function 'CGUIControlGroup::GetFirstFocusableControl' declared here: different number of parameters (1 vs 0)
  virtual CGUIControl *GetFirstFocusableControl(int id);
                       ^
1 warning generated.
```

Reply:
Oh, Visual Studio didn't show that.
Took me a while to think about it, but I'm not sure it would be a good idea to use the GUIControlGroup version of the function instead because it will recurse in its children (controlgroup/list in another controlgroup/list), which is pointless in this situation: the GUIWindow already triggered that recursion through GetFocusedControl to deliver the action.

We could add a parameter to the GUIControlGroup::GetFirstFocusableControl to skip recursion, or leave it like that and let it recurse for nothing (not sure if there are any examples in skins or settings)
Same thing for GetLastFocusableControl that could be added to GUIControlGroup instead.
While at it, GetFocusedControl could also have another parameter to stop recursion when called from the GUIControlGroupList.

Reply:
I'm not 100% sure, but you should be able to solve this via the `using` keyword:

```patch
--- a/xbmc/guilib/GUIControlGroupList.h
+++ b/xbmc/guilib/GUIControlGroupList.h
@@ -28,6 +28,8 @@ struct SGUIControlAndOffset
 class CGUIControlGroupList : public CGUIControlGroup
 {
 public:
+  using CGUIControlGroup::GetFirstFocusableControl;
+
   CGUIControlGroupList(int parentID, int controlID, float posX, float posY, float width, float height, float itemGap, int pageControl, ORIENTATION orientation, bool useControlPositions, uint32_t alignment, const CScroller& scroller);
   ~CGUIControlGroupList(void) override;
   CGUIControlGroupList* Clone() const override { return new CGUIControlGroupList(*this); }
```

That makes the error go away, as both functions coincide in the class now.

Reply:
it's been a while, I'm going to freshen this up.

Reply:
I'm crossing my fingers that your suggestion removes the warning.
Extending the base class is a bit of a mess as it also is the base class of a lot of other classes + the function does a lot more than needed > I'd prefer ControlGroupList to have its own.